### PR TITLE
Full rewrite of entire NAT sec and some reordering

### PR DESCRIPTION
**Pull request by mirjak**

### CODE REVIEW

File: draft-ietf-quic-manageability.md [hunk 1]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
...
 
-## Passive Network Performance Measurement and Troubleshooting
+## Network Address Translation (NAT)
 
-Limited RTT measurement is possible by passive observation of QUIC traffic;
-see {{sec-rtt}}. No passive measurement of loss is possible with the present
-wire image. Extremely limited observation of upstream congestion may be
-possible via the observation of CE markings on ECN-enabled QUIC traffic.
+This section uses the colloquial term Network Address
+Translation (NAT) to mean NAPT (section 2.2 of
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:822]:
```suggestion
Translation (NAT) to mean NAPT (Section 2.2 of
```

File: draft-ietf-quic-manageability.md [hunk 2]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
...
+{{?RFC3022}}), which overloads several IP addresses to one IP address or to an
+IP address pool, as commonly deployed in carrier-grade NATs or residential NATs.
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:824]:
```suggestion
{{?RFC3022}}), which rewrites addresses on flows from multiple IP addresses to
allow sharing of a limited pool of addresses, as commonly deployed in
carrier-grade NATs or residential NATs.
```

File: draft-ietf-quic-manageability.md [hunk 3]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
...
+
+{{?RFC4787}} contains some guidance on building NATs to interact constructively
+with a wide range of applications. This section extends the discussion to QUIC.
+
+In summary, the advise is simple: devices that perform NAT should use the
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:829]:
```suggestion
In summary, the advice is simple: devices that perform NAT should use the
```

File: draft-ietf-quic-manageability.md [hunk 4]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
...
+QUIC connection ID to map their internal state but only use the 4-tuple of IP
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:830]: I don't know what "map their internal state means".  If it means that the NAT can use extra knowledge of QUIC to learn more about how things are operating, then I don't know what use that has unless it is applied.

I would remove this extra piece, on the basis that it is confusing and misleading.  Other text can cover the learnings.

File: draft-ietf-quic-manageability.md [hunk 5]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
...
+addresses and ports as they would do for other UDP traffic or other transports
+as well.
+
+As already recognised in section {#sec-stateful} the QUIC connection ID is not
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:834]:
```suggestion
As already recognised in {{sec-stateful}} the QUIC connection ID is not
```

File: draft-ietf-quic-manageability.md [hunk 6]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
...
+guaranteed to be constant over the life-time of a connection and a change of
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:835]:
```suggestion
guaranteed to be constant over the lifetime of a connection. A NAT that uses connection IDs as part of its mapping will lose those mappings when a change of connection ID occurs.
```

This was missing the bit where you explain that the NAT is using connection IDs somehow.

File: draft-ietf-quic-manageability.md [hunk 7]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
...
+connection ID would therefore lead to loss of state or creation of unnecessary
+new state. Both should be avoided and indicate that a NAT device does not need
+to be QUIC-aware.
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:838]: This is just repetition of a previous statement.

File: draft-ietf-quic-manageability.md [hunk 8]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
...
+
+Section  {#sec-stateful} also provides further recommendations on time-outs
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:840]:
```suggestion
{{sec-stateful}} also provides further recommendations on timeouts
```

File: draft-ietf-quic-manageability.md [hunk 9]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
...
+for stateful network functions such as NATs. For QUIC it is important that time-outs
+are not extremely short, however, the general recommendation as provided in
+{{RFC4787}} for UDP is also appropriate for QUIC and as such again not NAT
+function are not required to be QUIC-aware.
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:844]: This is getting quite repetitive.  Maybe you can just list things that don't work:

* Tracking connections based on connection ID (connection IDs change).
* Changing NAT timeouts based on detecting QUIC (QUIC doesn't care).
* Rewriting addresses to ensure stability (requires endpoint cooperation; enables amplification attacks).

File: draft-ietf-quic-manageability.md [hunk 10]
```diff
@@ -809,20 +816,161 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
 of state in the middle of a connection. Use of connection ID specifically
 discouraged for NAT applications.
 
-## Passive Network Performance Measurement and Troubleshooting
+## Network Address Translation (NAT)
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md:819]: This section is too long for me.  I started reviewing it, but I lost the thread.  I don't think that this is an improvement.

**Review by martinthomson — CHANGES_REQUESTED** [review]

File: draft-ietf-quic-manageability.md (hunk 10 above)
Comment (mirjak) [reply draft-ietf-quic-manageability.md:819]: I actually agree. I was just trying to keep all of the points in Martin Duke's original text (while removing redundancy). PR #272 is a proposal for a minimum text version. Maybe we can start from there and re-add things if needed.

**Pull request closed without merging**